cmd/flexvolume/cmd/init.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2017 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/rook/rook/pkg/util/sys"
+	"github.com/spf13/cobra"
+)
+
+var (
+	initCmd = &cobra.Command{
+		Use:   "init",
+		Short: "Initialize the volume plugin",
+		RunE:  initPlugin,
+	}
+)
+
+func init() {
+	RootCmd.AddCommand(initCmd)
+}
+
+func initPlugin(cmd *cobra.Command, args []string) error {
+
+	moduleName := "rbd"
+	hasSingleMajor := false
+	// check to see if the rbd kernel module has single_major support
+	out, err := exec.Command("modinfo", "-F", "parm", moduleName).Output()
+	if err == nil {
+		hasSingleMajor = sys.Grep(string(out), "^single_major") != ""
+	}
+
+	opts := []string{moduleName}
+	if hasSingleMajor {
+		opts = append(opts, "single_major=Y")
+	}
+
+	// load the rbd kernel module with options
+	// TODO: should this fail if modprobe fails?
```

Comment:
How about running `lsmod` and "greping" through it to check if the module "exists"?
I personally am for failing when the module doesn't exist, as without the rbd module no rbd.

Reply:
the thing is that if kubelet runs in a container, it would have to have host mounted the /sbin/module dir in order check (or load) the rbd kernel module. This kernel module is a one time process and it might already be done by the admin. So even if the kernel module doesnt exist, it will fail later on during the attach process. We have had discussion of this in the past here: https://github.com/kubernetes/kubernetes/issues/45190